mindsdb/integrations/handlers/airtable_handler/airtable_handler.py

```diff
@@ -1,211 +1,192 @@
-        response = requests.get(url, headers=headers)
-        response = response.json()
-        records = response['records']
+        if columns:
+            df = df[columns]
 
-        new_records = True
-        while new_records:
-            try:
-                if response['offset']:
-                    params = {"offset": response['offset']}
-                    response = requests.get(url, params=params, headers=headers)
-                    response = response.json()
+        if query.order_by:
+            sort_params = []
+            for order in query.order_by:
+                field_name = order.field.parts[-1]
+                direction = "asc" if order.direction.lower() == "asc" else "desc"
+                sort_params.append((field_name, direction))
+            df = df.sort_values(by=[field_name], ascending=(direction == "asc"))
 
-                    new_records = response['records']
-                    records = records + new_records
-            except Exception:
-                new_records = False
+        return df
 
-        rows = [record['fields'] for record in records]
-        globals()[self.connection_data['table_name']] = pd.DataFrame(rows)
+    def insert(self, query: ast.Insert):
+        columns = [col.name for col in query.columns]
+        data = [
+            {col: value for col, value in zip(columns, row)} for row in query.values
+        ]
+        for record in data:
+            self.table.create(record)
 
-        self.connection = duckdb.connect()
-        self.is_connected = True
 
-        return self.connection
+class AirtableHandler(APIHandler):
+    """
+    This handler handles connection and execution of the Airtable statements.
+    """
 
-    def disconnect(self):
-        """
-        Close any existing connections.
-        """
+    name = "airtable"
 
-        if self.is_connected is False:
-            return
+    def __init__(self, name: str, connection_data: Optional[dict] = None, **kwargs):
+        super().__init__(name)
 
-        self.connection.close()
+        self.connection_data = connection_data or {}
+        self.access_token = self.connection_data.get("access_token")
+        self.base_id = self.connection_data.get("base_id")
         self.is_connected = False
-        return self.is_connected
+        self.api = None
+        self.base = None
+        self.table_fields = {}
+        self._register_table("tables", self)
+
+    def connect(self):
+        if self.is_connected:
+            return self.api
+        try:
+            self.api = pyairtable.Api(self.access_token)
+            self.base = self.api.base(self.base_id)
+            tables_metadata = self.base.tables()
+            for table_meta in tables_metadata:
+                table_name = table_meta.name
+                self._register_table(
+                    table_name, AirtableTable(self, table_name=table_name)
+                )
+            self.is_connected = True
+        except Exception as e:
```

Comment:
Maybe we can check for more specific errors also as Authentication issues here

Reply:
Yeah, it sounds like a good idea for debugging and system monitoring. I checked the SDK and was quite surprised that they don't have an authentication exception: https://pyairtable.readthedocs.io/en/stable/api.html#module-pyairtable.exceptions

I will add this one: `pyairtable.exceptions.PyAirtableError`